python_modules/dagster/dagster/_core/definitions/result.py

```diff
@@ -1,7 +1,7 @@
 from typing import NamedTuple, Optional, Sequence
 
 import dagster._check as check
-from dagster._annotations import PublicAttr
+from dagster._annotations import PublicAttr, experimental
 from dagster._core.definitions.asset_check_result import AssetCheckResult
 from dagster._core.definitions.data_version import DataVersion
 
```

Comment:
<!--__GRAPHITE_HTML_TAG_START__--><i class='graphite__hidden'>[Re: lines 15 to 15]<br/><br/></i>

<!--__GRAPHITE_HTML_TAG_END__-->looking at this let's add the base class. I think `MaterializeResult` and `ObserveResult` can just trivially inherit

```python
class AssetResult(NamedTuple(...)):
   # all the stuff

class MaterializeResult(AssetResult):
  pass

class ObserveResult(AssetResult):
  pass
```<!--__GRAPHITE_HTML_TAG_START__--><span class='graphite__hidden'><br/><br/>See this comment inline on <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/17824?utm_source=unchanged-line-comment">Graphite</a>.</span>

<!--__GRAPHITE_HTML_TAG_END__-->